src/hotspot/share/opto/library_call.cpp

```diff
@@ -7776,8 +7776,18 @@ bool LibraryCallKit::inline_blackhole() {
   assert(callee()->is_empty(), "Should have been checked before: only empty methods here");
   assert(callee()->holder()->is_loaded(), "Should have been checked before: only methods for loaded classes here");
 
+  // Like GraphKit::insert_mem_bar, this code takes and produces control
+  // to pin the Blackhole in the graph. Unlike GraphKit::insert_mem_bar,
+  // this code neither uses nor produces memory, which avoids the unfortunate
+  // side effects like breaking the optimizations across the blackhole.
+
+  MemBarNode* mb = MemBarNode::make(C, Op_Blackhole);
```

Comment:
Once thing to clear if you decide to keep modeling it as `MemBar`: pass `AliasIdxTop` as `alias_idx` .

Reply:
OK, I need to understand why, though. Does passing `AliasIdxTop` sentinel value here protects us from accidentally doing memory merges over the Blackhole node that does not touch memory? Is that the idea, or there is some other reason for it?